README.md

```diff
@@ -52,3 +52,38 @@ fastify.get('/', {
   const { x, y, z } = req.body
 })
 ```
+
+## Type definition of FastifyRequest + TypeProvider
+```ts
+import {
+  FastifyReply,
+  FastifyRequest,
+  RawRequestDefaultExpression,
+  RawServerDefault,
+} from 'fastify';
+import { Type } from '@sinclair/typebox';
+import { RouteGenericInterface } from 'fastify/types/route';
+import { TypeBoxTypeProvider } from '@fastify/type-provider-typebox';
+
+export type FastifyRequestTypebox<TSchema> = FastifyRequest<
+  RouteGenericInterface,
+  RawServerDefault,
+  RawRequestDefaultExpression<RawServerDefault>,
+  TSchema,
+  TypeBoxTypeProvider
+>;
+
+export const CreateProductSchema = {
+  body: Type.Object({
+    name: Type.String(),
+    price: Type.Number(),
+  }),
+};
+
+export const CreateProductHandler = (
+  req: FastifyRequestTypebox<typeof CreateProductSchema>
```

Comment:
```suggestion
  req: FastifyRequestTypebox<Static<typeof CreateProductSchema>>
```

Reply:
In this case, I saw that it didn't work with the Static function because TypeboxProvider already does this internally, so it throws an error: 
`Type '{ body: TObject<{ name: TString; price: TNumber; }>; }' does not satisfy the constraint 'TSchema'.
  Property '$static' is missing in type '{ body: TObject<{ name: TString; price: TNumber; }>; }' but required in type 'TSchema'.ts(2344)`